src/enclave/rpc_sessions.h

```diff
@@ -23,28 +25,38 @@ namespace enclave
   class RPCSessions : public AbstractRPCResponder
   {
   private:
-    size_t max_open_sessions_soft = 1000;
-    size_t max_open_sessions_hard = 1100;
+    using ListenInterfaceID = std::string;
+
+    struct ListenInterface
+    {
+      size_t open_sessions;
+      size_t peak_sessions;
+      size_t max_open_sessions_soft;
+      size_t max_open_sessions_hard;
+    };
+    std::map<ListenInterfaceID, ListenInterface> listening_interfaces;
 
     ringbuffer::AbstractWriterFactory& writer_factory;
     ringbuffer::WriterPtr to_host = nullptr;
     std::shared_ptr<RPCMap> rpc_map;
     std::shared_ptr<tls::Cert> cert;
 
     SpinLock lock;
-    std::unordered_map<size_t, std::shared_ptr<Endpoint>> sessions;
+    std::unordered_map<
+      tls::ConnID,
+      std::pair<ListenInterfaceID, std::shared_ptr<Endpoint>>>
+      sessions;
     size_t sessions_peak;
 
-    // Upper half of sessions range is reserved for those originating from
+    // Negative sessions are reserved for those originating from
     // the enclave via create_client().
-    std::atomic<size_t> next_client_session_id =
-      std::numeric_limits<size_t>::max() / 2;
+    std::atomic<tls::ConnID> next_client_session_id = -1;
```

Comment:
Client sessions are now small negative numbers (counting down from `-1`), rather than large unsigneds (counting up from `MAX / 2`), making them more readable in the logs.